### PR TITLE
Vertical tabs

### PR DESCRIPTION
**Pull request by zivl**
hey, 
this is the **vertical tabs items** and the initial stuff I did yesterday with vertical tabs.
you can see the `<VerticalTabsItem>` on storybook

### CODE REVIEW

File: src/VerticalTabsItem/VerticalTabsItem.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    size: oneOf(['small', 'medium']),
+
+    /** Prefix Icon */
+    prefixIcon: node,
+
+    /** Suffix Icon */
+    suffixIcon: node,
+
+    /** When present, it specifies that a button should automatically get focus when the page loads. */
+    autoFocus: bool,
```
Comment (mosheKerbel) [inline src/VerticalTabsItem/VerticalTabsItem.js]: i don't see any mention on auto focus in the spec
https://app.zeplin.io/project/5c8f6f1f9f489505cac756dc/screen/5d3060870184ff8e0db1b202

File: src/VerticalTabsItem/VerticalTabsItem.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+    /** should the text get ellipsis with tooltip, or should it get broken into lines when it reaches the end of its container */
+    ellipsis: bool,
```
Comment (mosheKerbel) [inline src/VerticalTabsItem/VerticalTabsItem.js]: not sure what you try to achieve here, I don't see the spec talking about tooltip.

File: src/VerticalTabsItem/VerticalTabsItem.js [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+    /** Children - only single child is allowed here */
+    children: element,
+
+    /** Disabled */
+    disabled: bool,
+
+    /** Tooltip floating modifiers */
+    tooltipModifiers: object,
```
Comment (mosheKerbel) [inline src/VerticalTabsItem/VerticalTabsItem.js]: not sure what you try to achieve here, I don't see the spec talking about tooltip.

File: src/VerticalTabsItem/VerticalTabsItem.js [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    return (
+      <div
+        {...styles(
+          'root',
+          { disabled, action: type === 'title', suffixIcon },
+          this.props,
+        )}
+        tabIndex={tabIndex}
+        ref={ref => (this.innerComponentRef = ref)}
+        autoFocus={autoFocus}
```
Comment (mosheKerbel) [inline src/VerticalTabsItem/VerticalTabsItem.js]: div does not have autoFocus

File: src/VerticalTabsItem/VerticalTabsItem.js [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    suffixIcon: node,
+
+    /** When present, it specifies that a button should automatically get focus when the page loads. */
+    autoFocus: bool,
+
+    /** should the text get ellipsis with tooltip, or should it get broken into lines when it reaches the end of its container */
+    ellipsis: bool,
+
+    /** Children - only single child is allowed here */
+    children: element,
```
Comment (mosheKerbel) [inline src/VerticalTabsItem/VerticalTabsItem.js]: you should use `node`

`string` does not work well with `element`
it currently produce an error
```
Warning: Failed prop type: Invalid prop `children` of type `string` supplied to `VerticalTabsItem`, expected a single ReactElement.
```

File: src/VerticalTabsItem/VerticalTabsItem.uni.driver.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import { baseUniDriverFactory } from 'wix-ui-test-utils/base-driver';
+
+export const verticalTabsItemDriverFactory = base => {
+  return {
+    ...baseUniDriverFactory(base),
+
+    /** Get the current count */
+    getCountText: async () =>
```
Comment (mosheKerbel) [inline src/VerticalTabsItem/VerticalTabsItem.uni.driver.js]: ???

File: src/VerticalTabsItem/docs/index.story.js [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+            code({
+              compact: true,
+              source:
+                '<VerticalTabsItem type="action">Action</VerticalTabsItem>',
+            }),
+          ]),
+          columns([
+            description({
+              title: 'Vertical Tab Title',
+              text: 'Vertical tab item title',
```
Comment (mosheKerbel) [inline src/VerticalTabsItem/docs/index.story.js]: title should not get hover color
image

File: src/VerticalTabsItem/VerticalTabsItem.js (hunk 1 above)
Comment (zivl) [reply src/VerticalTabsItem/VerticalTabsItem.js]: right... i thought its part of the `withFocus` HOC.
so no need?

File: src/VerticalTabsItem/VerticalTabsItem.js (hunk 2 above)
Comment (zivl) [reply src/VerticalTabsItem/VerticalTabsItem.js]: right. I'll remove

File: src/VerticalTabsItem/VerticalTabsItem.js (hunk 3 above)
Comment (zivl) [reply src/VerticalTabsItem/VerticalTabsItem.js]: will remove

File: src/VerticalTabsItem/VerticalTabsItem.js (hunk 5 above)
Comment (zivl) [reply src/VerticalTabsItem/VerticalTabsItem.js]: oh i've not noticed that...

File: src/VerticalTabsItem/VerticalTabsItem.uni.driver.js (hunk 6 above)
Comment (zivl) [reply src/VerticalTabsItem/VerticalTabsItem.uni.driver.js]: i did not do any testing what so ever, i guess its part of the generated code

File: src/VerticalTabsItem/docs/index.story.js (hunk 7 above)
Comment (zivl) [reply src/VerticalTabsItem/docs/index.story.js]: right.

File: src/VerticalTabsItem/VerticalTabsItem.js (hunk 1 above)
Comment (mosheKerbel) [reply src/VerticalTabsItem/VerticalTabsItem.js]: you can take a look at 2 examples of `withFocus` usage
https://github.com/wix/wix-style-react/blob/cad34db0b604713082047b98c409b5edbeb44d7a/src/StatisticsWidget/StatisticsItem/StatisticsItem.js#L154

https://github.com/wix/wix-style-react/blob/f3e6e7a20c4ec961cff9bad37bacc26dec143a3b/src/Stepper/Step/Step.js#L130

no `autoFocus`
but i think you shoud postpone the focus issues for later no?
we said we first want to make sure we have a basic working impl.

**Pull request merged**